tools/checky/src/checks/mod_use_order.rs

```diff
@@ -0,0 +1,184 @@
+use std::{cmp::Ordering, path::Path};
+
+use proc_macro2::LineColumn;
+use syn::{spanned::Spanned, File, Item, ItemMod, ItemUse, UsePath, UseTree, Visibility};
+
+use crate::output::output_diagnostic;
+
+pub fn check<P>(file_path: P, buffer: &str, file: &File) -> bool
```

Comment:
Should this be a public trait all checks have to implement?

Reply:
Additionally, combined with the proposal of a `Diagnostic` type. The return could be something like `Option<Vec<Diagnostic>>`